### PR TITLE
allow user to specify tls version for backward compatibility

### PR DESCRIPTION
**Pull request by pureiboi**
* [x] I added CHANGELOG entry for this change.
* [ ] Change is not relevant to the end user.

## Changes

resolve issue https://github.com/thanos-io/thanos/issues/7624

- expose flag to allow user to input tls version
- when flag is not provided, default tls version to 1.3
## Verification

sslscan is used to verify the tls version that was enabled 

tls 1 3

tls 1 2

### CODE REVIEW

File: cmd/thanos/config.go [hunk 1]
```diff
@@ -45,6 +46,9 @@ func (gc *grpcConfig) registerFlag(cmd extkingpin.FlagClause) *grpcConfig {
 	cmd.Flag("grpc-server-tls-client-ca",
 		"TLS CA to verify clients against. If no client CA is specified, there is no client verification on server side. (tls.NoClientCert)").
 		Default("").StringVar(&gc.tlsSrvClientCA)
+	cmd.Flag("grpc-server-tls-min-version",
+		"TLS minimum version to gRPC server, unset will default to tls 1.3, allow values: [\"1.0\", \"1.1\", \"1.2\", \"1.3\"]").
```
Comment (harry671003) [inline cmd/thanos/config.go:50]: Maybe: 

"TLS supported minimum version for gRPC server. If no version is specified, it'll default to 1.3. Allowed values: [[\"1.0\", \"1.1\", \"1.2\", \"1.3\"]]"

**Review by harry671003 — APPROVED** [review]
LGTM! Thank you.

File: cmd/thanos/config.go (hunk 1 above)
Comment (pureiboi) [reply cmd/thanos/config.go]: I have updated the description. also remove the extra square bracket. from [["1.0", "1.1", "1.2", "1.3"]] to ["1.0", "1.1", "1.2", "1.3"], thought that might be typo and confusing for reading.

**Review by fpetkovski — APPROVED** [review]

**Comment by pureiboi** [comment]
resolved conflict, and merged latest from main.

requires approval

**Comment by pureiboi** [comment]
@fpetkovski if you can help to review again

**Review by MichaHoffmann — DISMISSED** [review]

**Review by GiedriusS — APPROVED** [review]

**Pull request merged**